toolkit/scripts/cherry_pick.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+set -e
+
+function help {
+    echo "Cherry-pick commit to a specific branch and create Github PR"
+    echo "Usage:"
+    echo "[MANDATORY] -r REPOSITORY -> name of the repository in the format <owner>/<repo-name>"
+    echo "[MANDATORY] -p PR_NUMBER -> number of the PR to cherry-pick commit from"
+    echo "[MANDATORY] -t TARGET_BRANCH -> target branch to cherry-pick commit to"
+}
+
+function collect_pr_info {
+    repo=$1
+    pr_number=$2
+
+    commit_hash=$(gh pr view $pr_number --repo $repo --json mergeCommit --jq '.mergeCommit.oid')
+    original_pr_title=$(gh pr view $pr_number --repo $repo --json title --jq '.title')
+}
+
+function cherry_pick {
+    commit_hash=$1
+    target_branch=$2
+    repo=$3
+    pr_number=$4
+    original_pr_title=$5
+
+    tmp_branch="cherry-pick-$target_branch-$commit_hash"
+
+    echo "Cherry picking commit ($commit_hash) to target branch $target_branch"
+
+    # reset the current working tree to clean state
+    git reset --hard
+    git clean -df
+    git checkout -- .
+
+    # create a temporary branch from target branch to perform cherry pick
+    git fetch --all
+    git checkout -b "$tmp_branch" origin/"$target_branch"
+
+    if ! git cherry-pick -x "$commit_hash"; then
+        echo "Cherry pick failed. Displaying conflicts below"
+        git diff --diff-filter=U
+        gh pr comment "$pr_number" \
+            --repo "$repo" \
+            --body "Cherry-pick failed for branch \`$target_branch\`. See run logs for more details: $RUN_URL"
+    else
+        echo "pushing to remote"
+        git push -u origin "$tmp_branch"
+        echo "done pushing to remote"
```

Comment:
Nit-pick: could we unify the logging convention? It looks like capital letters and no period at the end seems to be the most popular in this script.
```suggestion
        echo "Pushing to remote"
        git push -u origin "$tmp_branch"
        echo "Done pushing to remote"
```